pages/marketplace/new.tsx

```diff
@@ -74,7 +73,10 @@ export default function New() {
   const [form] = Form.useForm();
   const [pendingAddress, setPendingAddress] = useState<string>();
   const { setVisible } = useWalletModal();
-  const { solana, wallet, looking } = useContext(WalletContext);
+  const solana = useWallet();
+  const { publicKey } = solana;
+  const pubkey = publicKey?.toBase58();
```

Comment:
similarly, `pubkey` is pretty ambiguous IMO. Better would be like `userWalletPubKey` or something